kola/tests/ignition/oem.go

```diff
@@ -75,13 +106,62 @@ storage:
 `),
 	})
 	register.Register(&register.Test{
-		Name:        "cl.ignition.oem.reuse",
-		Run:         reusePartition,
+		// Check new behavior from https://github.com/flatcar/bootengine/pull/58
+		// to not have to specify the OEM filesystem in Ignition
+		Name:        "cl.ignition.oem.indirect",
+		Run:         reusePartitionNew,
+		ClusterSize: 1,
+		Distros:     []string{"cl"},
+		// This test overwrites the grub.cfg which does not work on cloud environments after reboot
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 3603},
+		UserData: conf.Butane(`---
+variant: flatcar
+version: 1.0.0
+storage:
+  files:
+    - path: /oem/grub.cfg
+      mode: 0644
+      overwrite: true
+      contents:
+        inline: |
+          set linux_append="flatcar.autologin"
+          # Needed if --qemu-skip-mangle is not set
+          set linux_console="console=ttyS0,115200"
+`),
+	})
+	register.Register(&register.Test{
+		Name:        "cl.ignition.oem.reuse.old",
+		Run:         reusePartitionOld,
 		ClusterSize: 1,
 		Distros:     []string{"cl"},
 		// This test overwrites the grub.cfg which does not work on cloud environments after reboot
 		Platforms:  []string{"qemu", "qemu-unpriv"},
 		MinVersion: semver.Version{Major: 2983},
+		EndVersion: semver.Version{Major: 3603},
+		// Using CLC format here also covers the ign-converter case
+		UserData: conf.ContainerLinuxConfig(`storage:
+  filesystems:
+     - name: oem
+       mount:
+         device: "/dev/disk/by-label/OEM"
+         format: "ext4"
+  files:
+    - path: /grub.cfg
+      filesystem: oem
+      mode: 0644
+      contents:
+        inline: |
+          set linux_append="flatcar.autologin"`),
+	})
+	register.Register(&register.Test{
```

Comment:
Why is this duplicated? I don't see a difference (on the first look, maybe I miss something)?